### DISCUSSION TITLE
Long coding on the Flipper Zero

### DISCUSSION THREAD

**Discussion by 0x000006** [MatthewKuKanich/CAN_Commander#3] [Ideas]
Will it be supported on the Flipper Zero? If yes, will there be also a long coding helper?

**Comment by MatthewKuKanich** [comment]
Yes, it will have full Flipper Zero support. The app is still in development but almost complete. I'm not sure what you mean by long coding, could you explain? If you're referring to extended CAN frames then yes.

**Comment by 0x000006** [comment]
Long coding a.k.a. coding allows you to change some bits in the car's control unit. For example you can change a bit that enables a secret feature like rolling down your windows with your key fob.

**Comment by tehrabbitt** [comment]
So this is the thing I was talking about a while back in regards to VW/
Audi with VCDS.  We should look into adding long coding support + a wizard
with pre-saved coding for common things like enable/disable DRL, set
voltage of DRL (to prolong bulb life), increase voltage of headlights
during night operation (reduced bulbsm life but brighter), automatic wipers
on rain detection, open / close sunroof with key fob, etc...

**Reply by MatthewKuKanich** [reply]
I've been researching and adding compatability in the program. Since these codes are different from typical CAN messages we will use a separate file type to save to an SD card similar to DBC files. This is a very car specific feature so testing is difficult but in the works
